spec/Section 6 -- Execution.md

```diff
@@ -734,9 +734,21 @@ the {objectValue}.
 
 ResolveAbstractType(abstractType, objectValue):
 
-- Return the result of calling the internal method provided by the type system
-  for determining the Object type of {abstractType} given the value
-  {objectValue}.
+- Let {resolvedType} be the result of calling the internal method provided by
+  the type system for determining the Object type of {abstractType} given the
+  value {objectValue}.
+- Assert {resolvedType} is an Object type.
+- Assert {IsSubType(abstractType, resolvedType)}.
+- Return {resolvedType}.
+
+IsSubType(abstractType, objectType):
+
+- If {abstractType} is an Interface type.
+  - If {objectType} implements {abstractType}, return {true}.
+  - Otherwise, return {false}.
+- If {abstractType} is a Union type.
+  - If {objectType} is a member type of {abstractType}, return {true}.
+  - Otherwise, return {false}.
```

Comment:
Based on the [discussion here](https://github.com/graphql/graphql-spec/pull/973#discussion_r932110074) and careful reading of IsValidImplementationFieldType, I believe we can just use that existing algorithm:

```suggestion
- Assert {resolvedType} is not the same type as {abstractType}.
- Assert {IsValidImplementationFieldType(resolvedType, abstractType)}.
- Return {resolvedType}.
```

It may make sense to extract the relevant parts into a new algorithm as @yaacovCR suggests though, as the name of the algorithm isn't necessarily the most sensible name for this usage.

Reply:
Actually I'm going to dismiss this; the proposed implementation is clearer.